src/Parser/Parser.hs

```diff
@@ -217,6 +216,19 @@ reservedOp = P.reservedOp lexer
 comma :: ParsecT String u Identity String
 comma = P.comma lexer
 
+-- | A parser for a literal expression
+literal :: Parser (Expr SourcePos)
+literal =
+  I <$> int
+  <|>
+  B <$> (reserved "True" *> pure True)
+  <|>
+  B <$> (reserved "False" *> pure False)
+  <|>
+  S <$> capIdentifier
+  <|>
+  Tuple <$> parens ((:) <$> (lexeme literal <* lexeme comma) <*> commaSep literal)
```

Comment:
This has the capacity to match `(1,)` if I am reading this correctly. Maybe it will be caught elsewhere, but by definition our tuples are a pair or more in BoGL. We could add 
```
... (lexeme literal <* lexeme comma <* lexeme literal) <*> try $ lexeme comma *> commaSep literal
```
Or something of that form, basically guarantee a tuple to parse only if it has minimum two elements, and potentially look for others preceded by an additional comma.

Reply:
A test against this would be revealing if this is a problem in the first place actually.

Reply:
You're right. Good catch.